### PR TITLE
add forum-id

### PR DESCRIPTION
**Pull request by itta611**
When you want to reply to someone else's post, by clicking the link, you can add the postID with a link at the beginning of the post.

Tested on chrome 92.0.4515.159（Official Build） （x86_64）
スクリーンショット 2021-09-04 午前8 41 04
preview:
スクリーンショット 2021-09-04 午前8 46 38

### CODE REVIEW

**Comment by LankyBox01** [comment]
I don't think you should directly modify transifex JSON...
image

File: addons-l10n/ja/forum-id.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"forum-id/@name":"フォーラムID", "forum-id/@description":"返信のためのIDを投稿の先頭に追加します。", "forum-id/add-btn": "IDを追加"}
```
Comment (RedGuy12) [inline addons-l10n/ja/forum-id.json:1]: no

File: addons/forum-id/userscript.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+export default async function ({addon, console, msg}) {
+  await addon.tab.waitForElement(".postfootright > ul", {
+    markAsSeen: true,
+  });
+  const buttons = document.querySelectorAll('.postfootright');
+  buttons.forEach(function (elm) {
+    let lineNode = document.createTextNode('| ');
+    let addBtn = document.createElement('li');
+    elm.querySelector('.postreport').prepend(lineNode);
+    addBtn.innerHTML = `<a href="#reply">${msg('add-btn')}</a>`;
```
Comment (RedGuy12) [inline addons/forum-id/userscript.js:10]: no

File: addons/forum-id/userscript.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+export default async function ({addon, console, msg}) {
+  await addon.tab.waitForElement(".postfootright > ul", {
+    markAsSeen: true,
+  });
+  const buttons = document.querySelectorAll('.postfootright');
+  buttons.forEach(function (elm) {
+    let lineNode = document.createTextNode('| ');
```
Comment (RedGuy12) [inline addons/forum-id/userscript.js:7]: sharedspace

**Comment by WorldLanguages** [comment]
Can you update the PR body so that it's clearer what the addon does? With screenshots if possible.

**Comment by itta611** [comment]
Fix bug:
Fixed an issue where users couldn't add an ID when they were already typing a post.

**Comment by itta611** [comment]
Any update?

**Comment by WorldLanguages** [comment]
Can someone who uses the forums frequently (idk, @CST1229?) confirm whether this would be useful? Why not just quote the post?

**Comment by CST1229** [comment]
> 
> 
> Can someone who uses the forums frequently (idk, @CST1229?) confirm whether this would be useful? Why not just quote the post?

I feel that simply automatically adding a link to the post when quoted would be more useful.

**Comment by itta611** [comment]
> I feel that simply automatically adding a link to the post when quoted would be more useful.
That would make the post larger when the quote is quoted, so I think it's cleaner to use the post ID

**Comment by itta611** [comment]
> Also, it doesn't work for me on Firefox 92...
> kép

Fixed. review please.

**Comment by mxmou** [comment]
> > I feel that simply automatically adding a link to the post when quoted would be more useful.
> 
> That would make the post larger when the quote is quoted, so I think it's cleaner to use the post ID

Making posts a few pixels larger (the link can be small and people should snip long quotes anyway) isn't a big problem. Some people already add links to their quotes:
image

I think using post numbers instead of quotes is only common in the Japanese forum - could the button use post numbers there and quotes in other forums? It could be called something like "Reply with link".

**Comment by itta611** [comment]
I see. In Japan, we often use only IDs without quotes, so I'll add an "Add ID link" button and an "Add ID link with quote" button.

**Comment by itta611** [comment]
@WorldLanguages @CST1229 @mxmou @RedGuy12 @LankyBox01
Update

**Comment by WorldLanguages** [comment]
For this to be merged it needs a few changes
- Don't edit addon-l10n/ja/* directly
- `forum-id` goes at the bottom of addons.json
- Use `document.createElement` instead of `innerHTML`
- Use the forumsBeforePostReport space with [addon.tab.appendToSharedSpace](https://scratchaddons.com/docs/reference/addon-api/addon.tab/addon.tab.appendtosharedspace/)

**Comment by itta611** [comment]
@WorldLanguages fixed.

File: addons/forum-id/userscript.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    let addBtnAElement = document.createElement('a');
+    addBtnAElement.href = '#reply';
+    addBtnAElement.textContent = msg('add-btn');
+    addBtn.appendChild(addBtnAElement);
+    addBtn.addEventListener('click', addIDLink);
+    addon.tab.appendToSharedSpace({space: 'forumsBeforePostReport', element: addBtn, scope: elm});
+  });
+  function addIDLink(e) {
+    let textEditor = document.querySelector('.markItUpEditor');
+    let idName = e.path[6].querySelector('.box-head > .conr').textContent;
```
Comment (WorldLanguages) [inline addons/forum-id/userscript.js:17]: `e.path` will be undefined on Firefox

**Pull request merged**